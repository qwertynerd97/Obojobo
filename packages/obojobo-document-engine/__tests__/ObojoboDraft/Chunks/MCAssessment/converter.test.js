jest.mock('Common', () => ({
	Store: {
		getItemForType: () => ({
			slateToObo: jest.fn(),
			oboToSlate: jest.fn()
		})
	}
}))

import Converter from '../../../../ObojoboDraft/Chunks/MCAssessment/converter'
const SETTINGS_NODE = 'ObojoboDraft.Chunks.MCAssessment.Settings'
const CHOICE_LIST_NODE = 'ObojoboDraft.Chunks.MCAssessment.ChoiceList'
const MCCHOICE_NODE = 'ObojoboDraft.Chunks.MCAssessment.MCChoice'

describe('MCAssessment Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with no content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => null
			},
			nodes: [
				{
					type: 'NotADefinedNode'
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('slateToObo converts a Slate node to an OboNode', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return { responseType: 'pick-one-multiple-correct' }
				}
			},
			nodes: [
				{
					type: CHOICE_LIST_NODE,
					nodes: [
						{
							type: MCCHOICE_NODE,
							data: {
								get: () => {
									return { score: 100 }
								}
							}
						}
					]
				},
				{
					type: SETTINGS_NODE,
					nodes: {
						first: () => ({
							data: {
								get: () => 'pick-one-multiple-correct'
							}
						}),
						last: () => ({
							data: {
								get: () => false
							}
						})
					}
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('slateToObo converts a Slate node to an OboNode with two correct', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return { responseType: 'pick-one' }
				}
			},
			nodes: [
				{
					type: CHOICE_LIST_NODE,
					nodes: [
						{
							type: MCCHOICE_NODE,
							data: {
								get: () => {
									return { score: 100 }
								}
							}
						},
						{
							type: MCCHOICE_NODE,
							data: {
								get: () => {
									return { score: 100 }
								}
							}
						},
						{
							type: MCCHOICE_NODE,
							data: {
								get: () => {
									return { score: 0 }
								}
							}
						}
					]
				},
				{
					type: SETTINGS_NODE,
					nodes: {
						first: () => ({
							data: {
								get: () => 'pick-one'
							}
						}),
						last: () => ({
							data: {
								get: () => false
							}
						})
					}
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			children: [
				{
					type: MCCHOICE_NODE
				},
				{
					type: 'NotADefinedNode'
				}
			],
			content: {}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
