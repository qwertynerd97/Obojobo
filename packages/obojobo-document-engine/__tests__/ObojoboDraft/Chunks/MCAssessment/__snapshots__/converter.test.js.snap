// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MCAssessment Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {},
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "nodes": Array [
        undefined,
        undefined,
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.MCAssessment.ChoiceList",
    },
    Object {
      "nodes": Array [
        Object {
          "data": Object {
            "current": undefined,
            "display": "Response Type",
            "name": "responseType",
            "options": Array [
              "pick-one",
              "pick-all",
            ],
            "type": "type-select",
          },
          "isVoid": true,
          "object": "block",
          "type": "Parameter",
        },
        Object {
          "data": Object {
            "checked": undefined,
            "display": "Shuffle",
            "name": "shuffle",
            "options": undefined,
            "type": "type-toggle",
          },
          "isVoid": true,
          "object": "block",
          "type": "Parameter",
        },
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.MCAssessment.Settings",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`MCAssessment Converter slateToObo converts a Slate node to an OboNode 1`] = `
Object {
  "children": Array [
    undefined,
  ],
  "content": Object {
    "responseType": "pick-one",
    "shuffle": false,
  },
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`MCAssessment Converter slateToObo converts a Slate node to an OboNode with no content 1`] = `
Object {
  "children": Array [],
  "content": Object {},
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`MCAssessment Converter slateToObo converts a Slate node to an OboNode with two correct 1`] = `
Object {
  "children": Array [
    undefined,
    undefined,
    undefined,
  ],
  "content": Object {
    "responseType": "pick-one-multiple-correct",
    "shuffle": false,
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
